### ISSUE TITLE
git.latest reports deleted tags in remote repo as new_tags in changes

### ISSUE THREAD

**Issue by dosercz** [saltstack/salt#48299]
After deleting tag deleted-tag in github repo (deleted-tag is present in local git repo) git.latest produced these output:
```
          ID: repo
    Function: git.latest
        Name: git@github.com:me/repo.git
      Result: True
     Comment: Repository /home/me/repo is up-to-date
     Started: 17:02:53.334089
    Duration: 4084.445 ms
     Changes:
              ----------
              new_tags:
                  - deleted-tag
```
It's strange and confusing that this is reported as new_tags, because this is not newly added. It was there before and was only deleted in remote repo. I have not seen this change with older salt versions.
IMO this tag should be deleted in this case to keep repo up-to-date.

### Setup
```
repo:
  git.latest:
    - name: git@github.com:me/repo.git
    - rev: master
    - target: /home/me/repo
    - force_checkout: True
    - force_fetch: True
    - force_reset: True
```
### Versions Report
```
Salt Version:
           Salt: 2018.3.1

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.5.3
      docker-py: Not Installed
          gitdb: 2.0.0
      gitpython: 2.1.1
          ioflo: Not Installed
         Jinja2: 2.9.4
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.24.0
           Mako: 1.0.6
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: 1.3.7
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.13 (default, Nov 24 2017, 17:33:09)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: 2.0.1
        timelib: Not Installed
        Tornado: 4.4.3
            ZMQ: 4.2.1

System Versions:
           dist: debian 9.4
         locale: UTF-8
        machine: x86_64
        release: 4.9.0-6-amd64
         system: Linux
        version: debian 9.4
```

**Comment by garethgreenaway** [comment]
@terminalmage Thoughts?

**Comment by terminalmage** [comment]
This is because the current code is just using a symmetric difference of the two sets of tags. I've corrected that and am writing a test for this case now.

**Comment by terminalmage** [comment]
Fixed in #49305.
